sdk/pom.xml

```diff
@@ -30,6 +30,9 @@
   <properties>
     <maven.deploy.skip>false</maven.deploy.skip>
     <grpc.version>1.39.0</grpc.version>
+    <argLine>
+      --add-opens java.base/java.util=ALL-UNNAMED
```

Comment:
Is this needed for tests only? If so, is it possible to only enable this for tests?

Reply:
It is documented in the [surefire:test](https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#argLine) documentation. The property belongs to Surefire and I would argue it is - as a consequence - restricted to test phase. I am not aware of any other plugin using it.

Reply:
This property should be set in surefire itself, not in the general <argLine>. Also see this migration guide for Java 17 (longer term, but might as wel fix it now ) regarding JEP 403 - Strongly Encapsulate JDK internals.

https://github.com/johanjanssen/JavaUpgrades#jep-403-strongly-encapsulate-jdk-internals

From that page:
```
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-surefire-plugin</artifactId>
    <configuration>
        <argLine>
            --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED
        </argLine>
    </configuration>
</plugin>
```

Reply:
> [...] who knows what other plugins might pick it up otherwise.

The JaCoCo plugin, for once, works with that property. The _shortest_ way to make Surefire and JaCoCo play nicely together is to define a property: `<argLine>-your -extra -arguments</argLine>`.

The JaCoCo plugin also shows a more verbose example, which is to add this to the Surefire plugin (as @TomCools and Johan Janssen suggest):

```xml
<argLine>
    ${argLine}
    --add-opens java.base/java.lang=ALL-UNNAMED
    --add-opens java.base/java.util=ALL-UNNAMED
</argLine>
```

Since we have no evidence that it _unintentionally_ influences other plugins, I would argue to stick with the `<argLine>--add-opens...</argLine>` approach.

Reply:
Fair enough, let's be pragmatic about this. Consider my remark closed. 😉